src/EmbraceUncertainty.jl

```diff
@@ -123,7 +130,45 @@ function readme()
     return Markdown.parse_file(joinpath(CACHE[], "README.txt"))
 end
 
-export GENRES
+"""
+    tagpad(v::AbstractVector{<:Integer}, tag::String="S"; pool::Bool=true)
+
+Convert `v` to a vector of strings prepended with `tag` and padded to a constant string length.
+If `pool` is `true`, the default, the resulting vector of strings is converted to a `PooledArray`.
+
+The reason for padding the numeric strings is so that the strings sort lexicographically in the
+same order as the original numeric values.
+
+The single-argument version, e.g. `tagpad(:I)`, returns a partially-applied function that can be used
+in a `transform` or `select` call.
+
+```@example
+show(tagpad(repeat(1:10, inner=2)))
+```
+"""
+function tagpad(v::AbstractVector{<:Integer}, tag::String="S"; pool::Bool=true)
+    tagged = string.(tag, lpad.(v, maximum(ndigits, v), '0'))
```

Comment:
if you really wanted to, you could add one more method here

```julia
tagpad(v::Integer, tag::AbstractString, padlen::Integer) = string(tag, lpad(v, padlen, '0'))
```

Reply:
Good suggestion.  Thanks.